fraud/bad_encoding.go

```diff
@@ -0,0 +1,179 @@
+package fraud
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/tendermint/tendermint/pkg/consts"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	pb "github.com/celestiaorg/celestia-node/fraud/pb"
+	"github.com/celestiaorg/celestia-node/service/header"
+)
+
+type BadEncodingProof struct {
+	BlockHeight uint64
+	// Shares contains all shares from row/col
+	// Shares that did not pass verification in rmst2d will be nil
+	// For non-nil shares MerkleProofs are computed
+	Shares []*Share
+	// Position represents the number of row/col where ErrByzantineRow/ErrByzantineColl occurred
+	Position uint8
+	isRow    bool
+}
+
+func CreateBadEncodingFraudProof(
```

Comment:
Should have doc comments. Maybe even a runnable [example test](https://go.dev/blog/examples) to explain the usage/flow here better.

Reply:
Added comment